### ISSUE TITLE
Cannot Assembly.LoadFrom() in net5.0-windows from the .exe 

### ISSUE THREAD

**Issue by jbennink** [dotnet/sdk#14759]
As part of program startup we check several .dll and .exe files in the main executable folder and load the assembly and initialize some interface types. This worked fine in net48 but .exe files do not work in net5.0-windows and give a `System.BadImageFormatException` specifiying the `Bad IL format. The format of the file 'filename.exe' is invalid.'`

.DLL files seem to work fine. I can't find anything about this but it feels like a change not mentioned in the docs

Code inserted here and downloadlink to small sample code:
```
        // I kept some of the 'boilerplate' in this method but excluded everything after the failing line (21)
        public static void RegisterAssembly(string assemblyFilename)
        {
            string fullAssemblyFilename = "";
            if (!Path.IsPathRooted(assemblyFilename))
            {
                string directory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
                fullAssemblyFilename = Path.Combine(directory, assemblyFilename);
            }
            Assembly assembly = Assembly.UnsafeLoadFrom(fullAssemblyFilename);
        }

        [STAThread]
        static void Main()
        {
            // P.S. this also fails for different .exe than the current .exe running!
            RegisterAssembly("WindowsFormsApp2.exe");

            ...
        }
```

Sample code: [here](https://calway-my.sharepoint.com/:u:/g/personal/j_bennink_calway_nl/Ec12rWbF3uZCk0A-edMS2LYBiT68hSwyolXTRtnU_fpARg?e=Xe4Wif)

**Comment by jbennink** [comment]
UPDATE: Seems net5.0 compiles a .dll and a .exe. Perhaps I should use the .dll and the .exe is only a loader wrapper? If I point it to the .dll from the main project that generates the .exe it seems t owork. Still need to check when other code I left out in the sample works.

**Comment by PathogenDavid** [comment]
> Perhaps I should use the .dll and the .exe is only a loader wrapper?

Correct. As of .NET Core the generated exe is just a thin apphost. The actual .NET assembly is the dll.

**Comment by jespergustinmsft** [comment]
When using single-file publish, the DLL is no longer extracted (used to be in .NET Core 3), so how is one supposed to Assembly.LoadFrom() a single-file published exe? In .NET Core 3, we could use a (pretty ugly) workaround to call the exe with a hidden argument to get the folder in %temp% where the DLL was extracted to and then do another call to the DLL in that folder, but since this is no longer the case in .NET 5(?), what is the solution here?
